### PR TITLE
Improve docs on how to install logcli from source

### LINKED ISSUE #5108: Docs feedback: /getting-started/logcli.md
## Instructions for installing from source aren't clear/correct

running 
```
go get github.com/grafana/loki/cmd/logcli
```
gives me 
```
github.com/grafana/loki/cmd/logcli imports
	github.com/grafana/loki/pkg/logcli/query imports
	github.com/grafana/loki/pkg/loki imports
	github.com/cortexproject/cortex/pkg/querier/frontend imports
	google.golang.org/grpc/naming: cannot find module providing package google.golang.org/grpc/naming
go: warning: github.com/gomodule/redigo@v2.0.0+incompatible: retracted by module author: Old development version not maintained or published.
go: warning: k8s.io/client-go@v12.0.0+incompatible: retracted by module author: pre-module versions are obsolete
go: to switch to the latest unretracted version, run:
	go get <module>@latest
```

Then I try adding the `@latest` tag
```
go get github.com/grafana/loki/cmd/logcli@latest
```
which gives me 
```
github.com/grafana/loki/cmd/logcli imports
	github.com/grafana/loki/pkg/logcli/query imports
	github.com/grafana/loki/pkg/loki imports
	github.com/cortexproject/cortex/pkg/querier/frontend imports
	google.golang.org/grpc/naming: cannot find module providing package google.golang.org/grpc/naming
go: warning: github.com/gomodule/redigo@v2.0.0+incompatible: retracted by module author: Old development version not maintained or published.
go: warning: k8s.io/client-go@v12.0.0+incompatible: retracted by module author: pre-module versions are obsolete
go: to switch to the latest unretracted version, run:
	go get <module>@latest
```

my final guess was 
```
go install github.com/grafana/loki/cmd/logcli@latest
```
alas
```
go: downloading github.com/grafana/loki v1.6.1
go install: github.com/grafana/loki/cmd/logcli@latest (in github.com/grafana/loki@v1.6.1):
	The go.mod file for the module providing named packages contains one or
	more replace directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.
```

and running

### PR DESCRIPTION
**Pull request by chaudum**
**What this PR does / why we need it**:

Instead of mentioning the `go get` (or `go install`) command to install `logcli` from source (which does not work, see #5108), include the steps for how to build `logcli` from the checked out Github source.

**Which issue(s) this PR fixes**:
Closes #5108

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Review by kavirajk — APPROVED** [review]
LGMT

File: docs/sources/getting-started/logcli.md [hunk 1]
```diff
@@ -15,12 +15,16 @@ queries against a Loki instance.
 Download the `logcli` binary from the
 [Loki releases page](https://github.com/grafana/loki/releases).
 
-### From source
+### Compile from source
```
Comment (KMiller-Grafana) [inline docs/sources/getting-started/logcli.md:18]:
```suggestion
### Build logcli from source
```

File: docs/sources/getting-started/logcli.md [hunk 2]
```diff
@@ -15,12 +15,16 @@ queries against a Loki instance.
...
 
-Install `logcli` to your `$GOPATH/bin`:
+Clone Github repository and compile `logcli` from source:
```
Comment (KMiller-Grafana) [inline docs/sources/getting-started/logcli.md:20]:
```suggestion
Clone the Loki repository and build `logcli` from source:
```

File: docs/sources/getting-started/logcli.md [hunk 3]
```diff
@@ -15,12 +15,16 @@ queries against a Loki instance.
...
 
-```
-go get github.com/grafana/loki/cmd/logcli
+```bash
+git clone https://github.com/grafana/loki.git
+cd loki
+make logcli
+# optionally, move the binary into a directory that is part of $PATH
+cp cmd/logcli/logcli /usr/local/bin/logcli
```
Comment (KMiller-Grafana) [inline docs/sources/getting-started/logcli.md:27]:
```suggestion
```bash
git clone https://github.com/grafana/loki.git
cd loki
make logcli
```

Optionally, move the binary into a directory that is part of your `$PATH`:

```bash
cp cmd/logcli/logcli /usr/local/bin/logcli
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]

**Review by sandeepsukhani — APPROVED** [review]
LGTM

**Review by KMiller-Grafana — APPROVED** [review]
Thanks for the changes.  LGTM.

**Pull request merged**